src/createPersistor.js

```diff
@@ -27,6 +27,7 @@ export default function createPersistor (store, config) {
 
   // initialize stateful values
   let lastState = stateInit
+  let lastWritten = {} // Last written value, per key
```

Comment:
why do we need to store lastWritten? that is more or less what lastState tries to track.